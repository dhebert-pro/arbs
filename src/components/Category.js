import React from "react";

const Category = ({ slug }) => <div>La cat√©gorie est : {slug}</div>;

export default Category;