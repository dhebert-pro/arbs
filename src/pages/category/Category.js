import React from "react";
import withRedux from "next-redux-wrapper";

import { store } from "store/configureStore";
import Footer from "components/Footer";
import Category from "components/Category";
import Header from "components/Header";
import App from "components/App";

const Page = ({ slug }) => (
    <App className='App Homepage'>
        <div>
            <Header displaySearch={true} />
            <Category slug={slug} />
            <Footer />
        </div>
    </App>
);

export default withRedux(store)(Page);