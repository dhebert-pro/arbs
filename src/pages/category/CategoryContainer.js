import Category from "./Category";

Category.getInitialProps = ({ query }) => {
  return { slug: query.slug };
};

export default Category;
