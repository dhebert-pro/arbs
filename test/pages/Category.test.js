/*global jest expect test describe:true */
/*eslint no-undef: "error"*/
import React from "react";
import renderer from "react-test-renderer";
import Category from "pages/category/Category";
import CategoryContainer from "pages/category";

jest.mock("components/App", () => ({ children }) => (
  <div name="App">{children}</div>
));
jest.mock("components/Header", () => () => <div name="Header" />);
jest.mock("components/Footer", () => () => <div name="Footer" />);
jest.mock("components/CategorySlug", () => () => <div name="CategorySlug" />);

describe("Category Page", () => {
  test("should be rendered", done => {
    const component = renderer.create(<Category />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });
});

describe("Category Container", () => {
  test("should be rendered", done => {
    const component = renderer.create(<CategoryContainer />);
    expect(component).toMatchSnapshot();
    done();
  });
  test("should init slug", done => {
    const props = { query: { slug: "TEST" } };
    CategoryContainer.getInitialProps(props).then(result => {
      expect(result.initialProps).toEqual({ slug: "TEST" });
      done();
    });
  });
});
