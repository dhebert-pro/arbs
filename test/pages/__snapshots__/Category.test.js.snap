// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Container should be rendered 1`] = `
<div
  name="App"
>
  <div>
    <div
      name="Header"
    />
    <div
      name="CategorySlug"
    />
    <div
      name="Footer"
    />
  </div>
</div>
`;

exports[`Category Page should be rendered 1`] = `
<div
  name="App"
>
  <div>
    <div
      name="Header"
    />
    <div
      name="CategorySlug"
    />
    <div
      name="Footer"
    />
  </div>
</div>
`;
